functions/src/resolve-market.ts

```diff
@@ -15,156 +15,146 @@ import {
 } from '../../common/payouts'
 import { removeUndefinedProps } from '../../common/util/object'
 import { LiquidityProvision } from '../../common/liquidity-provision'
+import { APIError, newEndpoint, validate } from './api'
+
+const bodySchema = z.object({
+  outcome: z.enum(RESOLUTIONS),
+  contractId: z.string(),
+  value: z.number().optional(),
+  probabilityInt: z.number().gte(0).lt(100).optional(),
+  resolutions: z.map(z.string(), z.number()).optional(),
+})
+
+export const resolvemarket = newEndpoint(['POST'], async (req, auth) => {
+  const { outcome, value, contractId, probabilityInt, resolutions } = validate(
+    bodySchema,
+    req.body
+  )
+  const userId = auth.uid
+  if (!userId) throw new APIError(403, 'Not authorized')
+
+  const contractDoc = firestore.doc(`contracts/${contractId}`)
+  const contractSnap = await contractDoc.get()
+  if (!contractSnap.exists)
+    throw new APIError(404, 'No contract exists with the provided ID')
+  const contract = contractSnap.data() as Contract
+  const { creatorId, outcomeType, closeTime } = contract
+
+  if (outcomeType === 'FREE_RESPONSE') {
+    if (
+      isNaN(+outcome) &&
+      !(outcome === 'MKT' && resolutions) &&
+      outcome !== 'CANCEL'
+    )
+      throw new APIError(400, 'Invalid free response outcome')
+  } else if (outcomeType === 'NUMERIC') {
+    if (isNaN(+outcome) && outcome !== 'CANCEL')
+      throw new APIError(400, 'Invalid numeric outcome')
+  }
```

Comment:
I think it would be better to handle these validations with additional `zod` schemas applied conditional on the `outcomeType`, like in `placebet`.

Reply:
I took another pass at this which removed most of these validations